### PR TITLE
Add chrome trace format for recording samples

### PR DESCRIPTION
**Pull request by andrewjcg**
This adds "chrometrace" as a new format for the "record" command, which serializes samples using chrome trace events: https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU

### CODE REVIEW

File: src/chrometrace.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
+// This code is taken from the flamegraph.rs from rbspy
+// https://github.com/rbspy/rbspy/tree/master/src/ui/flamegraph.rs
```
Comment (benfred) [inline src/chrometrace.rs:2]: I don't see any chrometrace functionality in rbspy - is this license preamble accurate?

File: src/main.rs [hunk 2]
```diff
@@ -1,9 +1,12 @@
+#![feature(entry_insert)]
+
```
Comment (benfred) [inline src/main.rs:2]:
```suggestion
```

File: src/chrometrace.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+*/
+
+use std::cmp::min;
+use std::collections::HashMap;
+use std::io::Write;
+use std::time::Instant;
+
+use anyhow::Error;
+use serde_derive::Serialize;
+use serde_json;
```
Comment (benfred) [inline src/chrometrace.rs:36]:
```suggestion
```

**Review by benfred — APPROVED** [review]
thanks for this! overall it looks good - have some minor comments below

File: src/chrometrace.rs (hunk 1 above)
Comment (andrewjcg) [reply src/chrometrace.rs:2]: Ahh, sorry, I originally copied file from flamegraph.rs -- will fix.

Comment (benfred) [reply src/chrometrace.rs]: thanks! 

btw, there is an open issue in rbspy adding this functionality https://github.com/rbspy/rbspy/issues/140 =)

**Pull request merged**